system_tests/monitoring.py

```diff
@@ -175,3 +177,145 @@ def test_create_and_delete_metric_descriptor(self):
+            while True:
+                try:
+                    group.delete()
+                    break
+                except NotFound:
+                    if backoff_intervals:
+                        time.sleep(backoff_intervals.pop(0))
+                    else:
+                        raise
+
+    def test_create_group(self):
+        client = monitoring.Client()
+        group = client.group(
+            display_name=self.DISPLAY_NAME,
+            filter_string=self.FILTER,
+            is_cluster=self.IS_CLUSTER,
+        )
+        group.create()
+        self.to_delete.append(group)
+        self.assertTrue(group.exists())
+
+    def test_list_groups(self):
+        client = monitoring.Client()
+        new_group = client.group(
+            display_name=self.DISPLAY_NAME,
+            filter_string=self.FILTER,
+            is_cluster=self.IS_CLUSTER,
+        )
+        before_groups = client.list_groups()
+        before_names = set(group.name for group in before_groups)
+        new_group.create()
+        self.to_delete.append(new_group)
+        self.assertTrue(new_group.exists())
+        after_groups = client.list_groups()
+        after_names = set(group.name for group in after_groups)
+        self.assertEqual(after_names - before_names,
+                         set([new_group.name]))
+
+    def test_reload_group(self):
+        client = monitoring.Client()
+        group = client.group(
+            display_name=self.DISPLAY_NAME,
+            filter_string=self.FILTER,
+            is_cluster=self.IS_CLUSTER,
+        )
+        group.create()
+        self.to_delete.append(group)
+        group.filter = 'resource.type = "aws_ec2_instance"'
+        group.display_name = 'locally changed name'
+        group.reload()
+        self.assertEqual(group.filter, self.FILTER)
+        self.assertEqual(group.display_name, self.DISPLAY_NAME)
+
+    def test_update_group(self):
+        NEW_FILTER = 'resource.type = "aws_ec2_instance"'
+        NEW_DISPLAY_NAME = 'updated'
+
+        client = monitoring.Client()
+        group = client.group(
+            display_name=self.DISPLAY_NAME,
+            filter_string=self.FILTER,
+            is_cluster=self.IS_CLUSTER,
+        )
+        group.create()
+        self.to_delete.append(group)
+
+        group.filter = NEW_FILTER
+        group.display_name = NEW_DISPLAY_NAME
+        group.update()
+
+        after = client.fetch_group(group.id)
+        self.assertEqual(after.filter, NEW_FILTER)
+        self.assertEqual(after.display_name, NEW_DISPLAY_NAME)
+
+    def test_group_members(self):
+        client = monitoring.Client()
+        group = client.group(
+            display_name=self.DISPLAY_NAME,
+            filter_string=self.FILTER,
+            is_cluster=self.IS_CLUSTER,
+        )
+        group.create()
+        self.to_delete.append(group)
+
+        for _ in group.members():
+            pass    # Not necessarily reached.
```

Comment:
Gotcha. Maybe a simple `assertIsInstance`? Printing won't do much for the system test (and might get swallowed by the test runner?)